packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/AppTests.kt

```diff
@@ -292,42 +301,122 @@ class AppTests {
+            app.close()
+        }
+    }
+
+    @Test
+    fun encryptedMetadataRealm_openWithWrongKeyThrows() {
+        // Create new test app with a random encryption key
+        val correctKey = TestHelper.getRandomKey()
+        val app = TestApp(
+            appName = TEST_APP_FLEX,
+            builder = {
+                it
+                    .encryptionKey(correctKey)
+                    .syncRootDirectory("${appFilesDirectory()}/foo")
+            }
+        )
+
+        try {
+            // Create Realm in order to create the sync metadata Realm
+            val user = app.createUserAndLogin()
+            val syncConfig = SyncConfiguration
+                .Builder(user, setOf(ParentPk::class, ChildPk::class))
+                .build()
+            Realm.open(syncConfig).close()
+
+            // Create a configuration pointing to the metadata Realm for that app
+            val metadataDir = "${app.configuration.syncRootDirectory}/mongodb-realm/${app.configuration.appId}/server-utility/metadata/"
+            val wrongKey = TestHelper.getRandomKey()
+            val config = RealmConfiguration
+                .Builder(setOf())
+                .name("sync_metadata.realm")
+                .directory(metadataDir)
+                .encryptionKey(wrongKey)
+                .build()
+            assertTrue(fileExists(config.path))
+
+            // Open the metadata realm file with an invalid encryption key
+            assertNotEquals(correctKey, wrongKey)
+            assertFailsWith<IllegalArgumentException> { Realm.open(config) }
+        } finally {
+            app.close()
+        }
+    }
+
+    @Test
+    fun encryptedMetadataRealm_openWithoutKeyThrows() {
+        // Create new test app with a random encryption key
+        val app = TestApp(
+            appName = TEST_APP_FLEX,
+            builder = {
+                it
+                    .encryptionKey(TestHelper.getRandomKey())
+                    .syncRootDirectory("${appFilesDirectory()}/foo")
+            }
+        )
+
+        try {
+            // Create Realm in order to create the sync metadata Realm
+            val user = app.createUserAndLogin()
+            val syncConfig = SyncConfiguration
+                .Builder(user, setOf(ParentPk::class, ChildPk::class))
+                .build()
+            Realm.open(syncConfig).close()
+
+            // Create a configuration pointing to the metadata Realm for that app
+            val metadataDir = "${app.configuration.syncRootDirectory}/mongodb-realm/${app.configuration.appId}/server-utility/metadata/"
+            val config = RealmConfiguration
+                .Builder(setOf())
+                .name("sync_metadata.realm")
+                .directory(metadataDir)
+                .build()
+            assertTrue(fileExists(config.path))
+
+            // Open the metadata realm file without a valid encryption key
+            assertFailsWith<IllegalArgumentException> { Realm.open(config) }
```

Comment:
Maybe assert the exception message to be sure it's the expected encryption exception and not another unrelated throw

Reply:
Agreed, thanks :)